pkg/server/handler/v1alpha1/template.go

```diff
@@ -45,7 +54,59 @@ func ListTemplates(ctx context.Context, tenant string, includePublic bool, query
 		items = append(items, publicTemplates.Items...)
 	}
 
-	size := int64(len(items))
+	results := []v1alpha1.Stage{}
+	if query.Filter == "" {
+		results = items
+	} else {
+		// Support multiple filters rules: name or alias, and type, separated with comma.
+		filterParts := strings.Split(query.Filter, ",")
+		filters := make(map[string]string)
+		for _, part := range filterParts {
+			kv := strings.Split(part, "=")
+			if len(kv) != 2 {
+				return nil, cerr.ErrorQueryParamNotCorrect.Error(query.Filter)
+			}
+
+			filters[kv[0]] = strings.ToLower(kv[1])
```

Comment:
Why convert to lowercase, we can't filter case sensitively if so ?

Reply:
It is hard to user to make sure the case in filter, so we only provide case insensitive filter.